src/v/pandaproxy/schema_registry/sharded_store.cc

```diff
@@ -110,6 +110,33 @@ ss::future<bool> sharded_store::upsert(
     co_return co_await upsert_subject(marker, sub, version, id, deleted);
 }
 
+ss::future<subject_schema> sharded_store::has_schema(
+  subject sub, schema_definition def, schema_type type) {
+    auto versions = co_await get_versions(sub, include_deleted::no);
+
+    if (validate(def(), type).has_error()) {
+        throw as_exception(invalid_subject_schema(sub));
+    }
+
+    std::optional<subject_schema> sub_schema;
+    for (auto ver : versions) {
+        try {
+            auto res = co_await get_subject_schema(
+              sub, ver, include_deleted::no);
+            if (def == res.definition) {
+                sub_schema.emplace(std::move(res));
+                break;
+            }
+        } catch (...) {
```

Comment:
Can this exception handler be more specific?  Guessing it's probably there to catch things like errors for fetching deleted versions

Reply:
I have the same question about catching more specific exception (or adding logging). After the issues we've seen with memory, I'm pretty wary of throwing away things like bad_alloc which seem to bubble up from std:: and seastar in various ways.